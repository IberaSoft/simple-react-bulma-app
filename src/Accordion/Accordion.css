.accordions .accordion:not(:last-child) {
    margin-bottom: 1.5rem
}

.accordions .accordion {
    display: flex;
    flex-direction: column;
    background-color: #f5f5f5;
    border-radius: 4px;
    font-size: 1rem
}

.accordions .accordion strong {
    color: currentColor
}

.accordions .accordion a:not(.button):not(.tag) {
    color: currentColor;
    text-decoration: underline
}

.accordions .accordion:not(:first-child) .accordion-header {
    border-radius: 0
}

.accordions .accordion:not(:last-child) {
    margin-bottom: 0
}

.accordions .accordion:not(:last-child) .accordion-content {
    border-radius: 0
}

.accordions .accordion .toggle,
.accordions .accordion [data-action=toggle] {
    cursor: pointer
}

.accordions .accordion .accordion-header {
    align-items: center;
    background-color: #4a4a4a;
    border-radius: 4px 4px 0 0;
    color: #fff;
    display: flex;
    justify-content: space-between;
    line-height: 1.25;
    padding: .5em .75em;
    position: relative
}

.accordions .accordion .accordion-header .toggle {
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    -moz-appearance: none;
    -webkit-appearance: none;
    background-color: rgba(10, 10, 10, .2);
    border: none;
    border-radius: 290486px;
    display: inline-block;
    flex-grow: 0;
    flex-shrink: 0;
    margin-left: .75em;
    height: 20px;
    width: 20px;
    max-height: 20px;
    max-width: 20px;
    min-height: 20px;
    min-width: 20px;
    outline: 0;
    position: relative;
    vertical-align: top
}

.accordions .accordion .accordion-header .toggle::before {
    color: #fff;
    display: block;
    left: 0;
    position: absolute;
    top: .2em;
    right: 0;
    bottom: 0;
    content: '\002B'
}

.accordions .accordion .accordion-header .toggle::after {
    display: none
}

.accordions .accordion .accordion-header+.accordion-body {
    overflow-y: hidden;
    max-height: 0;
    color: #4a4a4a;
    border: 1px solid #dbdbdb;
    border-radius: 4px;
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-top: none;
    transition: max-height .2s ease
}

.accordions .accordion .accordion-header+.accordion-body .accordion-content {
    padding: 1em 1.25em
}

.accordions .accordion.is-active .accordion-header .toggle::before {
    content: '\002D'
}

.accordions .accordion.is-active .accordion-body {
    max-height: 100em
}