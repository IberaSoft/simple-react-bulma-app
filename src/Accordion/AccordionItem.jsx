import React from 'react';

//css
import './AccordionItem.css';

const AccordionItem = ({ title, children, isActive, clickHeader }) => {
    let className = "accordion--item accordion--item-primary"

    if (isActive) className = `${className} accordion--item-is-active`

    return (
        <article className={className}>

            <div className="accordion--header" onClick={clickHeader}>
                <p>{title}</p>
                <button className="button--toggle"></button>
            </div>

            <div className="accordion--body">
                <div className="accordion--content">
                    {children}
                </div>
            </div>

        </article>
    );
}

export default AccordionItem;