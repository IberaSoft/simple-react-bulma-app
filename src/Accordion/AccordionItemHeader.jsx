import React from 'react';

const AccordionItemHeader = (props) => {
    return (
        <div className="accordion--header" onClick={props.onClick}>
            <p>{props.children}</p>
            <button className="button--toggle"></button>
        </div>
    )
}

export default AccordionItemHeader;