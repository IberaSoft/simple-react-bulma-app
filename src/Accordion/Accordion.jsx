import React, { Component } from 'react';

// Css
import './Accordion.css';

class Accordion extends Component {
    state = {
        activeIndex: 0
    }

    headerClick = (index) => {
        if (index !== this.state.activeIndex) this.setState({ activeIndex: index })
    }

    headerClass = (index) => (this.state.activeIndex === index) ? "is-active" : ""

    render() {
        const { accordionItems } = this.props
        const { activeIndex } = this.state

        return (
            <section className="accordions">
                {
                    accordionItems.map((tab, index) =>
                        <article key={tab.title + index}
                            onClick={() => this.headerClick(index)}
                            className={`accordion is-primary ${this.headerClass(index)}`}>
                            <div className="accordion-header toggle">
                                <p>{tab.title}</p>
                                <button className="toggle" aria-label="toggle"></button>
                            </div>
                            <div className="accordion-body">
                                <div className="accordion-content">
                                    {accordionItems[activeIndex].content}
                                </div>
                            </div>
                        </article>
                    )
                }
            </section>
        );
    }
}

export default Accordion;