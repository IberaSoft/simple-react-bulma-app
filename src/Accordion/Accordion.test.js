import React from 'react'
import Accordion from './Accordion'
import ShallowRenderer from 'react-test-renderer/shallow';

const defaultAccordionItems = [{
      title: 'Accordion 1',
      content: <p><strong>Content 1</strong> Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.</p >
    },
    {
      title: 'Accordion 2',
      content: <p><strong>Content 2</strong> Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.</p >
    },
    {
      title: 'Accordion 3',
      content: <p><strong>Content 3</strong> Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.</p >
    },
    {
      title: 'Accordion 4',
      content: <p><strong>Content 4</strong> Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.</p >
    },
  ];

describe('Accordion component', () => {
    it('Should match the snapshot', () => {
        const renderer = new ShallowRenderer();
        const result = renderer.render(
            <Accordion
                accordionItems={defaultAccordionItems}
                activeIndex={0}
                />);
        expect(result).toMatchSnapshot();
    });
})