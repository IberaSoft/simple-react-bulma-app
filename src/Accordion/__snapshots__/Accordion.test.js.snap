// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Should match the snapshot 1`] = `
<section>
  <AccordionItem
    clickHeader={[Function]}
    isActive={false}
    title="Accordion 1"
  >
    <p>
      <strong>
        Content 1
      </strong>
       Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.
    </p>
  </AccordionItem>
  <AccordionItem
    clickHeader={[Function]}
    isActive={false}
    title="Accordion 2"
  >
    <p>
      <strong>
        Content 2
      </strong>
       Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.
    </p>
  </AccordionItem>
  <AccordionItem
    clickHeader={[Function]}
    isActive={false}
    title="Accordion 3"
  >
    <p>
      <strong>
        Content 3
      </strong>
       Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.
    </p>
  </AccordionItem>
  <AccordionItem
    clickHeader={[Function]}
    isActive={false}
    title="Accordion 4"
  >
    <p>
      <strong>
        Content 4
      </strong>
       Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.
    </p>
  </AccordionItem>
</section>
`;
